base/src/consensus_state.rs

```diff
@@ -81,9 +86,129 @@ pub struct ConsensusState {
     pub supercharge_coinbase: bool,
 }
 
+/// For ConsensusConstants Default
+pub struct ConsensusConstants {
+    /// Point of finality (number of confirmations)
+    pub k: Length,
+    /// Number of slots per epoch
+    pub slots_per_epoch: Length,
+    /// No of slots in a sub-window = 7
+    pub slots_per_sub_window: Length,
+    /// Maximum permissable delay of packets (in slots after the current)
+    pub delta: Length,
+    /// Timestamp of genesis block in unixtime
+    pub genesis_state_timestamp: BlockTime,
+    /// Sub windows within a window
+    pub sub_windows_per_window: Length,
+    /// Number of slots before minimum density is used in chain selection
+    pub grace_period_end: Length,
+}
+
+impl Default for ConsensusConstants {
+    fn default() -> Self {
+        // TODO: read from config
+        Self {
+            k: Length(290),
+            slots_per_epoch: 7140.into(),
+            slots_per_sub_window: 7.into(),
+            delta: 0.into(),
+            genesis_state_timestamp: BlockTime(1615939200000),
+            sub_windows_per_window: 11.into(),
+            grace_period_end: Length(1440),
+        }
+    }
```

Comment:
where are these values from?

Reply:
Use in this [line](https://github.com/ChainSafe/mina-rs/blob/70098ebe162625ac8be18fec3c5da589fedd417b/consensus/tests/checkpoint.rs#L211) and this [line](https://github.com/ChainSafe/mina-rs/blob/70098ebe162625ac8be18fec3c5da589fedd417b/consensus/tests/checkpoint.rs#L51) is origin from ProrocolConstant. See [Discussion](https://github.com/ChainSafe/mina-rs/pull/93#discussion_r780879450)

Reply:
Fixed document [Initialize consensus JSON template](https://github.com/MinaProtocol/mina/blob/32f529d8d8d712a44ee75be66061ce08cbdc8924/docs/specs/consensus/README.md#61-initialize-consensus)